### ISSUE TITLE
Encoding array in HTTP.post data

### ISSUE THREAD

**Issue by otroboe** [meteor/meteor#5035]
Hello.

When I'm trying to use HTTP.post, I found that the method [_encodeParams](https://github.com/meteor/meteor/blob/devel/packages/url/url_common.js#L8) can't encode an array of objects.

I've created a [jsFiddle](https://jsfiddle.net/yxd4drt7/) to compare with the [$.param](http://api.jquery.com/jquery.param/) method of jQuery.

Is this a normal behavior for this method ?

Thanks

**Comment by stubailo** [comment]
@otroboe if you are talking about form encoding, I don't think there is a standard for how an array is supposed to be encoded in a POST request. I guess jQuery just picked one option and ran with it.

What is the use case here? Are you calling an external API that expects to be POSTed an array?

**Comment by otroboe** [comment]
I'm using a simple localhost API written in nodejs/express, and the POST params I send come from my form. I send the request by the server side for security reasons.

In the first place I was trying like this, and this is when I discover the 'issue' I posted:
```
    var options = {
        params: formData,
        headers: {
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
        }
    };
    HTTP.post(url, options, function (err, res) {
        // ...
    });
```
But I've found an other way, and It works well:
```
    var options = {
        data: formData
    };
    HTTP.post(url, options, function (err, res) {
        // ...
    });
```
But as you said it's not a real issue, but I've probably misunderstood the documentation
> params Object
> Dictionary of request parameters to be encoded and placed in the URL (for GETs) or request body (for POSTs). 

Thanks for your response :)

**Comment by tanmynguyen** [comment]
hi @otroboe . I use 
`headers: {
          'Content-type': "application/x-www-form-urlencoded"
        },
        data: {
          parmlist: "10"
        }`
but when response ` { 'cache-control': 'private',
 'content-length': '141',
**'content-type': 'text/html'**`
so I can't call result api??? I can't set headers in http.post :'(

**Comment by otroboe** [comment]
@tanmynguyen Sorry I don't use Meteor anymore, I can't help you :(

**Comment by abernix** [comment]
Sorry to hear you don't use Meteor anymore @otroboe.  I realize it's been a while, but I'm happy to say that this should be fixed in Meteor 1.4.3.1 thanks to @hwillson's PR!  Thanks!

    meteor update --release 1.4.3.1

The result is that `Array`s and `Object`s will be encoded in a way more similar to Rails, PHP and `jQuery.param`. 

For others encountering this problem: `Object`s were not previously serialized at all, but `Array`s were simply `join`-ed with commas.  If you're encountering a problem with the new encoding and you're passing an array (and expect it to be joined in this way), you should simply `[].join(',')` the array into `String` yourself, prior to passing it to `HTTP.call`.  For more information on this (and one problem which popped up internally), see https://github.com/meteor/meteor/pull/8261#issuecomment-278708655 and also: https://github.com/meteor/meteor/issues/5065.

Closing!

@tanmynguyen You're encountering an entirely different problem.  Please consult the [`HTTP.call` documentation](https://docs.meteor.com/api/http.html) for more information and consult the [Meteor forums](http://forums.meteor.com/) `help` section or [Stack Overflow tagged with `meteor`](https://stackoverflow.com/questions/tagged/meteor).
